torch/distributed/_tensor/device_mesh.py

```diff
@@ -34,12 +34,38 @@
 class _MeshEnv:
     def __init__(self) -> None:
         self.mesh_stack: List[DeviceMesh] = []
+        self.child_to_parent_mapping = {}
```

Comment:
> nit: should specify the type of this: Dict[torch.Tensor, DeviceMesh]

I was thinking of just doing a mapping of child_to_parent: Dict[DeviceMesh, DeviceMesh] since we have the reference of the child mesh anyway. Wdyt?

Reply:
Yeah I think that sounds good.

another follow up we should do (can be done in follow up PR): we should probably think how to clear the child_to_parent map, i.e. if the child mesh goes out of scope or be python GCed, we need to remove the entry in this map.